src/main/java/animals/service/FileService.java

```diff
@@ -0,0 +1,49 @@
+package animals.service;
+
+import animals.model.Node;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import java.io.File;
+import java.io.IOException;
+
+public class FileService {
+    public void save(Node root, String type, String fileName) {
+        ObjectMapper objectMapper = getObjectMapper(type);
+        try {
+            objectMapper
+                    .writerWithDefaultPrettyPrinter()
+                    .writeValue(new File(fileName), root);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public Node load(String type, String fileName) {
+        Node root = new Node();
+        ObjectMapper objectMapper = getObjectMapper(type);
+        try {
+            root = objectMapper.readValue(new File(fileName), Node.class);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return root;
+    }
+
+    private ObjectMapper getObjectMapper(String type) {
```

Comment:
Instead of if/else for each type we can have separate class for each type:
abstract FileService
XmlFileService
JsonFileService
YamlFileService
In this case we will not have `type` as argument in each method. And Game will receive one of this FileService and will not care about `type` as well